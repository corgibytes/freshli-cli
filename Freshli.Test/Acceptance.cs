using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using Freshli.Languages.Php;
using Freshli.Languages.Ruby;
using Freshli.Languages.Python;
using Xunit;

namespace Freshli.Test
{
  [UseReporter(typeof(XUnit2Reporter))]
  public class Acceptance
  {
    public Acceptance() {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      ManifestFinder.Register<PhpComposerManifestFinder>();
      ManifestFinder.Register<PipRequirementsTxtManifestFinder>();

      FileHistoryFinder.Register<GitFileHistoryFinder>();
    }

    private DateTime _testingBoundary =
      new DateTime(2020, 01, 01, 0, 0, 0, DateTimeKind.Utc);

    [Fact]
    public void RubyGemsWithGitHistory()
    {
      var runner = new Runner();

      var rubyFixturePath = Fixtures.Path("ruby", "nokotest");
      var results = runner.Run(rubyFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsWithHistoryViaGitHub() {
      var runner = new Runner();

      var repoUrl =
        "https://github.com/corgibytes/freshli-fixture-ruby-nokotest";
      var results = runner.Run(repoUrl, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsFeedbinHistoryViaGitHub()
    {
      var runner = new Runner();

      var repoUrl = "https://github.com/feedbin/feedbin";
      var results = runner.Run(repoUrl, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact] 
    public void RubyGemsClearanceHistoryViaGitHub()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();
      var results = runner.Run("https://github.com/thoughtbot/clearance");
      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void ComposerWithoutGitHistory()
    {
      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "large");
      var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void DrupalComposerWithoutGitHistory()
    {
      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "drupal");
      var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RequirementsTxtPyspider() {
      var runner = new Runner();

      var results = runner.Run(
        "https://github.com/binux/pyspider",
        asOf: _testingBoundary
      );

      Approvals.VerifyAll(results, "results");
    }
  }
}
