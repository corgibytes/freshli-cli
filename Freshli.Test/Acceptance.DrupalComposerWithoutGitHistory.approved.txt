results[0] = { Date: 2020-01-01T00:00:00, LibYear: { _packagesValues: [ { Name: "drupal/composer_autoloader", Version: "1.3.0", PublishedAt: 2018-03-11T00:00:00, Value: 3.213698630136986 }, { Name: "drupal/drupal", Version: "8.8.1", PublishedAt: 2019-12-18T00:00:00, Value: -0.42191780821917807 },  ], Total: 2.791780821917808 } }
